src/neo4j-schema/graphQLMapper.js

```diff
@@ -0,0 +1,289 @@
+import schema from './entities';
+import neo4jTypes from './types';
+import _ from 'lodash';
+
+const relationDirective = (relType, direction) =>
+  `@relation(name: "${relType}", direction: "${direction}")`;
+
+const mapOutboundRels = (tree, node, config) => {
+  const labels = node.getLabels();
+
+  return _.flatten(
+    labels.map(label => {
+      // Figure out which relationships are outbound from any label incident to
+      // this node.
+      const rels = tree.getRels().filter(rel => rel.isOutboundFrom(label));
+
+      return rels
+        .map(rel => {
+          const targetLabels = _.uniq(
+            _.flatten(rel.links.map(l => l.to))
+          ).sort();
+
+          if (targetLabels.length > 1) {
+            // This situation isn't handled yet, and arises when you have a setup like this:
+            // (:Customer)-[:BUYS]->(:Product)
+            // (:Customer)-[:BUYS]->(:Service)
+            // In this case, without type unions the destination type for :BUYS is ambiguous.
+            console.warn(
+              `RelID ${
+                rel.id
+              } for label ${label} has > 1 target type (${targetLabels}); skipping`
+            );
+            return null;
+          }
+
+          const tag = relationDirective(rel.getRelationshipType(), 'OUT');
+          const targetType = neo4jTypes.label2GraphQLType(targetLabels[0]);
+
+          const propName = rel.getGraphQLTypeName().toLowerCase();
+          const propNavigateToNode = `   ${propName}: [${targetType}] ${tag}\n`;
+
+          // If a relationship has props, we should always generate a type and field for
+          // it to provide access to those props.  If it doesn't have props, then only
+          // generate if user has told us to with the config.  Finally -- only univalents
+          // are supported ATM.
+          const shouldIncludeRelLink =
```

Comment:
It seems relationships with properties result in two GraphQL fields being created: one for the `@relation` field and a second for the relationship type. For example, from the Yelp data given OKAPI rel type properties:

```

relType            | propertyName       | propertyTypes     | mandatory
":`NARROWER_THAN`" | "score"            | ["Double"]        | true
```
currently generates two fields for the `NARROWER_THAN` relationship (`narrower_than` and `NARROWER_THAN_rel`):

```
type Category {
   _id: Long!
   name: String!
   narrower_than: [Category] @relation(name: "NARROWER_THAN", direction: "OUT")
   NARROWER_THAN_rel: [NARROWER_THAN]
   businesss: [Business] @relation(name: "IN_CATEGORY", direction: "IN")
}

type NARROWER_THAN @relation(name: "NARROWER_THAN") {
  from: Category!
  to: Category!
  score: Float!
}
```

Is that as intended? I would expect only the relationship type field (and not the `@relation` field`):

```
type Category {
   _id: Long!
   name: String!
   NARROWER_THAN_rel: [NARROWER_THAN]
   businesss: [Business] @relation(name: "IN_CATEGORY", direction: "IN")
}
```

I would only expect the `@relation` field to be included for relationships without properties.

Reply:
In autogen.js I'm checking in a change that sets schemaInferenceOptions.alwaysIncludeRelationships = false.   This will fix the behavior you're seeing.  

Basically -- it's configurable.   The default behavior is to only generate FOO_rel if the rel has properties.  I had left this toggled on in autogen.js so it was **always** generating it whether or not FOO_rel had any properties.

Per the design discussion, I don't think we can really guess ahead of time what the user wants so I think configurable is best.  The default is "false" I believe, so once I push the configuration change, you should see the behavior you prefer, which I think is a sensible default.

Reply:
OK sounds good.